library_generation/utils/utilities.py

```diff
@@ -216,9 +217,12 @@ def generate_prerequisite_files(
         if config.is_monorepo()
         else f"googleapis/{language}-{library_name}"
     )
-    api_id = (
-        library.api_id if library.api_id else f"{library.api_shortname}.googleapis.com"
-    )
+    if library.api_id == EMPTY_API_ID:
```

Comment:
Thanks for fixing this empty `api_id` issue! However, I think it was a mistake to have an empty `api_id` in repo-metadata.json, as it is specified in the [service yaml](https://github.com/googleapis/googleapis/blob/master/grafeas/v1/grafeas_v1.yaml#L3C7-L3C39). I created a [PR](https://github.com/googleapis/google-cloud-java/pull/10831) to fix it.

Reply:
Thanks, I didn't notice the api_id should not be empty.

I'll revert this change.

Reply:
Done.